flytekit/extend/backend/agent_service.py

```diff
@@ -17,37 +19,39 @@
 from flytekit.models.task import TaskTemplate
 
 
-class AgentService(AsyncAgentServiceServicer):
-    def CreateTask(self, request: CreateTaskRequest, context: grpc.ServicerContext) -> CreateTaskResponse:
+class AsyncAgentService(AsyncAgentServiceServicer):
+    async def CreateTask(self, request: CreateTaskRequest, context: grpc.ServicerContext) -> CreateTaskResponse:
         try:
             tmp = TaskTemplate.from_flyte_idl(request.template)
             inputs = LiteralMap.from_flyte_idl(request.inputs) if request.inputs else None
             agent = AgentRegistry.get_agent(context, tmp.type)
             if agent is None:
                 return CreateTaskResponse()
-            return agent.create(context=context, inputs=inputs, output_prefix=request.output_prefix, task_template=tmp)
+            return await asyncio.to_thread(
+                agent.create, context=context, inputs=inputs, output_prefix=request.output_prefix, task_template=tmp
```

Comment:
we should continue the async context downstream.

Maybe there is a better method, in the agent code
```python
class SnowflakeAgent:

def __init__():
   super().__init__(asynchronous=False)

def create():
   ...
```

And then here
async def CreateTask(...):
   if agent.asynchronous:
       return await agent.create(...)
   return asyncio.to_thread(agent.sync_create, ...)